src/stactools/package/stac.py

```diff
@@ -0,0 +1,179 @@
+        see https://pystac.readthedocs.io/en/latest/api.html#link
+        extent (list): Extent. ex [xmin, ymin, xmax, ymax].
+        See https://pystac.readthedocs.io/en/latest/api.html#spatialextent
+        start_time (datetime): Collection start time object- must be in UTC time zone
+        end_time (datetime): Collection end time object- must be in UTC time zone
+        href (str, optional): Location to store Collection json. Defaults to None,
+        and no json will be created.
+
+    Returns:
+        Collection: STAC Collection object
+    """
+    # The ID may be changed to be provided as an arg if desired
+    id = title.replace(" ", "-")
+
+    collection = Collection(
+        id=id,
+        title=title,
+        description=description,
+        providers=providers,
+        license=license,
+        extent=Extent(SpatialExtent([extent]),
+                      TemporalExtent([start_time, end_time])),
+        catalog_type=getattr(CatalogType, catalog_type),
+    )
+    collection.add_link(license_link)
+
+    if href:
+        collection.set_self_href(href)
+        collection.save_object()
+
+    return collection
+
+
+# This includes a set of minimum requirements for the creation of a STAC Item,
+# and should be added to/modified as necessary to develop a dataset-specific package.
+def create_item(
+    title: str,
+    description: str,
+    geometry: str,
+    start_datetime: datetime,
+    end_datetime: datetime = None,
+    asset_href: str = None,
+    asset_type: str = None,
+    projection_epsg: int = None,
+    href: str = None,
+) -> Item:
+    """Create a STAC Item
+
+    Args:
+        title (str): STAC Item name
+        description (str): STAC Item description
+        geometry (str): Footprint of the asset represented by this item, formatted
+        according to RFC 7946, section 3.1 (GeoJSON)
+        start_datetime (datetime): Asset datetime
+        end_datetime (datetime, optional): Asset end datetime if a range is required
+        asset_href (str, optional): Location of the asset represented by the item.
+        Defaults to None.
+        asset_type (str, optional): If an asset_href is provided, the asset type must
+        also be provided.
+        See https://pystac.readthedocs.io/en/latest/api.html#mediatype for the types.
+        Defaults to None.
+        projection_epsg (int, optional): EPSG code of the asset spatial reference.
+        Defaults to None.
+        href (str, optional): Location to store Item json. Defaults to None,
+        and no json will be created.
+
+    Raises:
+        ValueError: If an asset_href is provided, but no asset_type is provided
+
+    Returns:
+        Item: STAC Item object
+    """
+    # The ID may be changed to be provided as an arg if desired
+    id = title.replace(" ", "-")
+
+    properties = {
+        "title": title,
+        "description": description,
+    }
+
+    bbox = Polygon(geometry.get("coordinates")[0]).bounds
```

Comment:
It'd be nice to add a `stactools.core` utility that does this/similar things, since we do it a lot in dataset packages.

Reply:
https://github.com/stac-utils/stactools/issues/184